test/spec/displayrrender_spec.js

```diff
@@ -0,0 +1,69 @@
+describe('Displayr htmlwidgets', () => {
+    document.body.innerHTML = '<div>'+
+    '  <div id="upperwidget">'+
+    '    <div class="mockwidget" style="height: 50px" id="mockwidget-nodata">'+
+    '    </div>'+
+    '  </div>'+
+    '  <div id="lowerwidget">'+
+    '    <div class="mockwidget" style="height: 50px" id="mockwidget-bozo">'+
+    '      <script data-for="mockwidget-bozo" type="application/json">{"x": "bozo","evals":[],"jsHooks":{"render":""}}</script>'+
+    '    </div>'+
+    '  </div>'+
+    '</div>';
+
+    it('should be OK without data', () => {
+        const upper_div = document.querySelector('#upperwidget');
+        const nodata_widget = HTMLWidgets.instantiateDisplayrWidget(upper_div);
+        expect(nodata_widget).toBeDefined();
+        expect(nodata_widget.instance.alive).toEqual(true);
+        HTMLWidgets.destroyDisplayrWidget(nodata_widget);
+    });
+
+    it('should call renderValue() when data is provided', () => {
+        const lower_div = document.querySelector('#lowerwidget');
+        const bozo_widget = HTMLWidgets.instantiateDisplayrWidget(lower_div);
+        const text_in_svg = lower_div.querySelector('text').textContent;
+        expect(text_in_svg).toEqual('bozo', 'because that proves that renderValue() was called');
+        HTMLWidgets.destroyDisplayrWidget(bozo_widget);
+    });
+
+    it('should allow two widgets to be active at once', () => {
```

Comment:
I'd love to see a test where you add then remove a handful of widgets then at the end assert that the environment is as 'clean' as you'd expect - ie: collections empty, counts at zero, dom clean?